awx_collection/plugins/modules/organization.py

```diff
@@ -159,49 +167,65 @@ def main():
         association_fields['instance_groups'] = []
         for item in instance_group_names:
             association_fields['instance_groups'].append(module.resolve_name_to_id('instance_groups', item))
+    elif instance_group_names is None and enforced_defaults:
+        association_fields['instance_groups'] = []
```

Comment:
@AlanCoding would we be good with always assuming associations are `[]` by default? I think its probably find and I don't think we can extract it from the API anywhere but if we every did something like "auto adding" associations on new models this would break and have to be logic we apply in the modules themselves.

Reply:
If you make associations [] by default, it REMOVES all associations when submitted, which is the point of the enforced, and config as code. It is also a breaking change to how the modules work, unless we want enforced state to be default.

Reply:
Closing this one as it requires breaking changes.